src/main/java/duke/bin/UI.java

```diff
@@ -0,0 +1,97 @@
+package duke.bin;
+
+import duke.bin.common.Constants;
+import duke.bin.common.DukeException;
+import duke.bin.task.Task;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Handles output and input via terminal for the application
+ */
+public class UI {
+    private Scanner sc;
+
+    /**
+     * Default Constructor will instantiate a scanner object.
+     */
+    public UI () {
+        sc = new Scanner(System.in);
+    }
+
+    /**
+     * Displays the welcome message.
+     */
+    public void showWelcome() {
+        display("What can I do for you?", "Hello! I'm Duke");
+    }
+
+    /**
+     * Displays the error message to the user.
+     *
+     * @param exception exception that was thrown to be displayed
+     */
+    public void showError(DukeException exception) {
+        display(exception.getMessage());
+    }
+
+    /**
+     * Reads what the user has inputted.
+     *
+     * @return the input in String format
+     */
+    public String readCommand() {
```

Comment:
nextLine() function may throw exception when no line is found. So exception handling is recommended here.